modules/pipeline/initialize.go

```diff
@@ -191,12 +189,14 @@ func (p *provider) do() (*httpserver.Server, error) {
 	// aop
 	aop.Initialize(bdl, dbClient, reportSvc)
 
-	// engine start after all dependencies done
-	engine.Start()
-	// handle cron related after engine started
-	if err := doCrondAbout(crondSvc, pipelineSvc); err != nil {
-		return nil, err
-	}
+	p.ReconcilerElection.OnLeader(func(ctx context.Context) {
```

Comment:
How about degrade from leader but pipeline still running? You have to release leader functions.

Reply:
Use context.cancel or other methods.

Reply:
use ctx.Done channel leave leader func